pkg/schedulercache/resource_info.go

```diff
@@ -16,5 +16,70 @@ limitations under the License.
 
 package schedulercache
 
+import (
+	"fmt"
+	"math"
+
+	"k8s.io/api/core/v1"
+)
+
 type Resource struct {
+	MilliCPU float64
+	Memory   float64
+}
+
+func EmptyResource() *Resource {
+	return &Resource{
+		MilliCPU: 0,
+		Memory:   0,
+	}
+}
+
+func CopyResource(r *Resource) *Resource {
+	return &Resource{
+		MilliCPU: r.MilliCPU,
+		Memory:   r.Memory,
+	}
+}
+
+var minMilliCPU float64 = 10
+var minMemory float64 = 10 * 1024 * 1024
+
+func NewResource(rl v1.ResourceList) *Resource {
+	cpu := rl[v1.ResourceCPU]
+	mem := rl[v1.ResourceMemory]
+
+	return &Resource{
+		MilliCPU: float64(cpu.MilliValue()),
+		Memory:   float64(mem.Value()),
+	}
+}
+
+func (r *Resource) IsEmpty() bool {
+	return r.MilliCPU < minMilliCPU && r.Memory < minMemory
+}
+
+func (r *Resource) Add(rr *Resource) *Resource {
+	r.MilliCPU += rr.MilliCPU
+	r.Memory += rr.Memory
+	return r
+}
+
+func (r *Resource) Sub(rr *Resource) *Resource {
+	r.MilliCPU -= rr.MilliCPU
+	r.Memory -= rr.Memory
```

Comment:
we need to check the value of each resource, negative memory does not make sense.